### DISCUSSION TITLE
Feature Requests: Allow Custom Command Aliases

### DISCUSSION THREAD

**Discussion by JoshuaSarver-msr** [ankitpokhrel/jira-cli#431] [Ideas]
This is a really great tool, but on a day to day basis I'm only using a subset of the functionality available. Some of the command/flags feel unwieldy for a quick hop-in/hop-out session in `jira` to check a ticket, etc. And I'm lazy and like shortcuts. :)

I think adding a feature to create custom aliases would be highly beneficial. This could be done via `shell` aliases, but I having aliases integrated into the `jira` tool would be much nicer to use imo. Below is an example from a `Git` config file:

```
# ~/.gitconfig
[alias]
remote-url = config --get remote.origin.url
ac = !git add . && git commit -am
fp = ! git fetch && git pull

# list aliases
la = "!git config -l | grep alias | cut -c 7-"
```

In my case, one aliases (of several) I'd like to make is the following: 
```
# pseudo-code/config
me = issue list -a$(jira me)
```
Which would require executing just `jira me`.

If more details are needed please let me know and I'll fill them in!

**Comment by ankitpokhrel** [comment]
> And I'm lazy and like shortcuts :)

![Screen Shot 2022-08-14 at 11 33 16 AM](https://user-images.githubusercontent.com/2364546/184531004-403553e4-45ae-4e6f-8728-ca6e686316da.png)

Jokes aside, I think you are looking for a feature [like this](https://cli.github.com/manual/gh_alias). It would be a bit of a work and I am bit overwhelmed at the moment with all the requests. I am open to add this feature if someone is willing to work on and maintain this feature. For now, I think you'll have to live with shell aliases 🙃

```sh
alias jira-me = jira issue list -a$(jira me)
```

**Reply by JoshuaSarver-msr** [reply]
Thank you @ankitpokhrel !

**Comment by ChristophSchmidpeter** [comment]
I think it would be nice to be able to define aliases in way that e.g instead of always typing "issue" one can type simply "i", or whatever you set, which I cannot see how to do via shell aliases.

**Comment by ChristophSchmidpeter** [comment]
A great implementation example, in my opinion, is the [gh-cli alias command](https://cli.github.com/manual/gh_alias), which might serve as inspiration source.

**Reply by electriquo** [reply]
@ankitpokhrel it will be great if you could be inspired by gh-cli alias command. this will allow to share jira-cli configuration with others regardless of their shell and will eliminate shell alias conflicts

**Comment by bnadlerjr** [comment]
A workaround that I've been using to accomplish this is a custom bash function:

```
jira() {
    if [ "$1" = "my-issues" ]; then
        command jira issue list -a "$(jira me)"
    else
        command jira "$@"
    fi
}
```
